ledger/state_test.go

```diff
@@ -0,0 +1,464 @@
+	if err != nil {
+		t.Fatalf("failed to create chain manager: %v", err)
+	}
+
+	// Create LedgerState
+	ls := &LedgerState{
+		chain: cm.PrimaryChain(),
+		db:    db,
+	}
+
+	// Test GetBlock with empty hash
+	point := ocommon.NewPoint(12345, []byte{})
+	_, err = ls.GetBlock(point)
+	if err == nil {
+		t.Error("Expected error for empty hash, got nil")
+	}
+}
+
+// TestGetBlock_ZeroSlot tests GetBlock with slot 0 (genesis)
+func TestGetBlock_ZeroSlot(t *testing.T) {
+	// Setup test database
+	db, err := database.New(&database.Config{
+		BlobCacheSize: 1 << 20,
+		DataDir:       "",
+	})
+	if err != nil {
+		t.Fatalf("failed to create test database: %v", err)
+	}
+	defer db.Close()
+
+	// Create chain manager
+	cm, err := chain.NewManager(db, nil)
+	if err != nil {
+		t.Fatalf("failed to create chain manager: %v", err)
+	}
+
+	// Create genesis block
+	genesisBlock := models.Block{
+		Hash:     []byte{0x00, 0x11, 0x22, 0x33},
+		Slot:     0,
+		Number:   0,
+		PrevHash: []byte{0x00, 0x00, 0x00, 0x00},
+		Cbor:     []byte{0x82, 0x00, 0x81, 0x00},
+		Type:     5,
+	}
+
+	// Add block to chain
+	primaryChain := cm.PrimaryChain()
+	if err := primaryChain.AddBlock(genesisBlock, nil); err != nil {
+		t.Fatalf("failed to add genesis block: %v", err)
+	}
+
+	// Create LedgerState
+	ls := &LedgerState{
+		chain: primaryChain,
+		db:    db,
+	}
+
+	// Test GetBlock with slot 0
+	point := ocommon.NewPoint(0, genesisBlock.Hash)
+	result, err := ls.GetBlock(point)
+	if err != nil {
+		t.Fatalf("GetBlock failed for genesis block: %v", err)
+	}
+
+	if result.Slot != 0 {
+		t.Errorf("Expected slot 0 for genesis block, got %d", result.Slot)
+	}
+}
+
+// TestGetBlock_MultipleBlocks tests retrieving multiple different blocks
+func TestGetBlock_MultipleBlocks(t *testing.T) {
+	// Setup test database
+	db, err := database.New(&database.Config{
+		BlobCacheSize: 1 << 20,
+		DataDir:       "",
+	})
+	if err != nil {
+		t.Fatalf("failed to create test database: %v", err)
+	}
+	defer db.Close()
+
+	// Create chain manager
+	cm, err := chain.NewManager(db, nil)
+	if err != nil {
+		t.Fatalf("failed to create chain manager: %v", err)
+	}
+
+	primaryChain := cm.PrimaryChain()
+
+	// Create multiple test blocks
+	testBlocks := []models.Block{
+		{
+			Hash:     []byte{0x01, 0x00, 0x00, 0x00},
+			Slot:     100,
+			Number:   1,
+			PrevHash: []byte{0x00, 0x00, 0x00, 0x00},
+			Cbor:     []byte{0x82, 0x00, 0x81, 0x00},
+			Type:     5,
+		},
+		{
+			Hash:     []byte{0x02, 0x00, 0x00, 0x00},
+			Slot:     200,
+			Number:   2,
+			PrevHash: []byte{0x01, 0x00, 0x00, 0x00},
+			Cbor:     []byte{0x82, 0x00, 0x81, 0x00},
+			Type:     5,
+		},
+		{
+			Hash:     []byte{0x03, 0x00, 0x00, 0x00},
+			Slot:     300,
+			Number:   3,
+			PrevHash: []byte{0x02, 0x00, 0x00, 0x00},
+			Cbor:     []byte{0x82, 0x00, 0x81, 0x00},
+			Type:     5,
+		},
+	}
+
+	// Add blocks to chain
+	for _, block := range testBlocks {
+		if err := primaryChain.AddBlock(block, nil); err != nil {
```

Comment:
@coderabbitai This should be a "github.com/blinklabs-io/gouroboros/ledger/common".Block not a models.Block struct. You can use Decode() on models.Block to get a ledger.Block suitable for use here.